src/core/unit.h

```diff
@@ -757,6 +757,11 @@ typedef struct UnitVTable {
 extern const UnitVTable * const unit_vtable[_UNIT_TYPE_MAX];
 
 static inline const UnitVTable* UNIT_VTABLE(const Unit *u) {
+        assert(u);
+
+        if (u->type < 0 || u->type >= _UNIT_TYPE_MAX)
+                return NULL;
```

Comment:
this should be an assert()

Reply:
That just changes the crash from sigsegv to sigabrt, how does that fix the problem?